L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -90,24 +97,51 @@ void HGCalConcentratorSuperTriggerCellImpl::createAllTriggerCells(
 
 void HGCalConcentratorSuperTriggerCellImpl::assignSuperTriggerCellEnergyAndPosition(l1t::HGCalTriggerCell& c,
                                                                                     const SuperTriggerCell& stc) const {
+  //Compress and recalibrate STC energy
+  uint32_t compressed_value = getCompressedSTCEnergy(stc);
+
+  int thickness = 0;
+  if (triggerTools_.isSilicon(c.detId())) {
+    thickness = triggerTools_.thicknessIndex(c.detId(), true);
+  } else if (triggerTools_.isScintillator(c.detId())) {
+    thickness = 3;
```

Comment:
Actually, since this pattern of "check if silicon, use thickness index, otherwise for scintillator use pseudo index" is repeated various times in the trigger code, it seems like it should just all be absorbed into the `thicknessIndex()` tool function

Reply:
I followed your suggestion and absorbed this in the trigger tool function.